tool/src/org/antlr/v4/Tool.java

```diff
@@ -521,6 +521,19 @@ else if ( ruleAST==null ) {
 			// Make grammars depend on any tokenVocab options
 			if ( tokenVocabNode!=null ) {
 				String vocabName = tokenVocabNode.getText();
+				// Strip quote characters if any
+				int len = vocabName.length();
+				int firstChar = vocabName.charAt(0);
+				int lastChar = vocabName.charAt(len - 1);
+				if (len >= 2 && firstChar == '\'' && lastChar == '\'') {
+					vocabName = vocabName.substring(1, len-1);
+				}
```

Comment:
Not sure, what your regex does exactly, but it also strips the `'` from `'test` or `test'`. I think making sure that the we're dealing with a properly quoted string makes sense, i.e. that there's a quote character at the beginning and the end of the input.